ocaml/src/maintenance.ml

```diff
@@ -741,25 +753,29 @@ class client ?(retry_timeout = 60.)
                (if filter namespace_id
                 then
                   begin
-                    alba_client # nsm_host_access # get_namespace_info ~namespace_id
-                    >>= fun (_, devices, _) ->
-                    List.iter
-                      (fun osd_id ->
-                       if not (Hashtbl.mem threads osd_id)
-                       then begin
-                           Hashtbl.add threads osd_id ();
-                           Lwt.async
-                             (fun () ->
-                              Lwt_extra2.ignore_errors
+                    Lwt_pool2.use
```

Comment:
Push this throttling down to (or around) the `self # garbage_collect_device` call?